bibtex-actions.el

```diff
@@ -159,8 +154,18 @@ key associated with each one."
       ;;   3. the 'candidate-hidden' variable to be hidden
       (propertize candidate-main) " "
       (propertize candidate-suffix 'face 'bibtex-actions-suffix) " "
-      (propertize candidate-hidden 'invisible t)))
-    citekey))))
+      (propertize candidate-hidden 'invisible t)))))
+
+(defun bibtex-actions--advice (cands)
+  "The advice function to replace CANDS strings."
+  (cl-loop
+   for entry in cands
+   collect
+   ;; TODO the key logic
+   (setcar entry (bibtex-actions--transform-candidate entry))))
```

Comment:
Like, why can't I do something like this, @mohkale?

Doesn't work, but ...

Reply:
I don't see why you can't

```lisp
(advice-add #'bibtex-completion-candidates :filter-return #'bibtex-actions--advice)
```

but I don't see why you would. There's no benefit to doing this compared to just using 

```lisp
(cl-loop
   for entry in (bibtex-completion-candidates)
   collect
   ;; TODO the key logic
   (setcar entry (bibtex-actions--transform-candidate entry)))
```

And worse your altering the return format of a function in another library which means any other libraries aside from yours which also uses bibtex-completion might break.

Reply:
>  There's no benefit to doing this compared to just using ...

... what do you mean by your example @mohkale? Where would that `cl-loop` be used?

> And worse your altering the return format of a function in another library which means any other libraries aside from yours which also uses bibtex-completion might break.

Ah, hadn't thought about that. I guess that feature is better for local configs.

OK, then; bad idea :-)

Reply:
> what do you mean by your example @mohkale? Where would that cl-loop be used?

Wherever you have a call to `bibtex-completion-candidates` but want whatever your advice does to be applied to it before returning.

Reply:
@mohkale -  the only place I would call that function would be in the "candidates" let variable on `bibtex-actions-read`, and the only reason to make the mod would be so I can let bibtex-completion deal with caching of the candidates.

If you get a chance, if in fact you think that would work, could you fill out the example code please? I'm not understanding exactly how that would work.